File: src/main/java/io/github/aphelium/modules/coin/repositories/FiatRepository.java
package io.github.aphelium.modules.coin.repositories;

import io.github.aphelium.modules.coin.entities.Fiat;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface FiatRepository extends JpaRepository<Fiat, UUID> {

    Optional<Fiat> findByName(String name);

}

File: src/main/kotlin/io/github/aphelium/modules/coin/repositories/FiatRepository.kt
package io.github.aphelium.modules.coin.repositories

import io.github.aphelium.modules.coin.entities.Fiat
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface FiatRepository : JpaRepository<Fiat?, UUID?> {
    fun findByName(name: String?): Optional<Fiat?>
}